### PR TITLE
IBX-57: Exclusion category features

### PR DESCRIPTION
**Pull request by juskora**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-57
| Versions      | 4.0

### CODE REVIEW

File: docs/personalization/filters.md [hunk 1]
```diff
@@ -22,6 +22,20 @@ User profile-based filters are applicable in both publishing and eCommerce use c
 |Do not recommend items the user already consumed|The Personalization service stores the CONSUME events of every user for one year. When you activate this filter, the user will not be recommended the consumed content again.|
 |Max. repeated shows of identical recommendations per session|When you activate this filter and set a value, after a content item/product is recommended a certain number of times during the current user session, it is removed from all recommendation lists.|
 
+### Exclusions
+
+With Exclusion features you can exclude categories from the recommendation response by providing fixed category paths or by item category paths.
```
Comment (kaborkowski) [inline docs/personalization/filters.md:27]: Maybe could be `or by context item category paths`?

File: docs/personalization/filters.md [hunk 2]
```diff
@@ -22,6 +22,20 @@ User profile-based filters are applicable in both publishing and eCommerce use c
...
+
+!!! tip
+
+    You can use both options at the same time.
+
+- Exclude category of the context item - excludes items in the recommendation response from specified categories in the recommendation request. Select this checkbox if you want do not want to recommend a client items from the category of the currently rendered item. For example,
```
Comment (kaborkowski) [inline docs/personalization/filters.md:33]: I think that it coulde be something like:  `excludes items in the recommendation response from the same categories as context item`
too many: 'want'

**Review by kaborkowski — APPROVED** [review]

**Comment by juskora** [comment]
@dkorostynski could you review?

**Review by dkorostynski — APPROVED** [review]

**Review by kaborkowski — APPROVED** [review]

File: docs/personalization/filters.md [hunk 3]
```diff
@@ -22,6 +22,20 @@ User profile-based filters are applicable in both publishing and eCommerce use c
 |Do not recommend items the user already consumed|The Personalization service stores the CONSUME events of every user for one year. When you activate this filter, the user will not be recommended the consumed content again.|
 |Max. repeated shows of identical recommendations per session|When you activate this filter and set a value, after a content item/product is recommended a certain number of times during the current user session, it is removed from all recommendation lists.|
 
+### Exclusions
+
+With Exclusion features you can exclude categories from the recommendation response by providing fixed category paths or by context item category paths.
```
Comment (DominikaK) [inline docs/personalization/filters.md:27]: How about:
```suggestion
You can exclude categories from the recommendation response by providing fixed category paths or by context item category paths.
```
Unless the name "Exclusion" is important to retain?

File: docs/personalization/filters.md [hunk 4]
```diff
@@ -22,6 +22,20 @@ User profile-based filters are applicable in both publishing and eCommerce use c
...
+
+!!! tip
+
+    You can use both options at the same time.
```
Comment (DominikaK) [inline docs/personalization/filters.md:31]:
```suggestion
    You can use both fixed category and context item category paths at the same time.
```

File: docs/personalization/filters.md [hunk 5]
```diff
@@ -22,6 +22,20 @@ User profile-based filters are applicable in both publishing and eCommerce use c
...
+
+- Exclude category of the context item - excludes items in the recommendation response from the same categories as the context item in the recommendation request. Select this checkbox if you want do not want to recommend a client items from the category of the currently rendered item. For example,
```
Comment (DominikaK) [inline docs/personalization/filters.md:33]:
```suggestion
- Exclude category of the context item - excludes items in the recommendation response from the same categories as the context item in the recommendation request.
Select this checkbox if you do not want to recommend items from the category of the currently rendered item. For example, 
```

File: docs/personalization/filters.md [hunk 6]
```diff
@@ -22,6 +22,20 @@ User profile-based filters are applicable in both publishing and eCommerce use c
...
+### Exclusions
+
+With Exclusion features you can exclude categories from the recommendation response by providing fixed category paths or by context item category paths.
+
+!!! tip
+
+    You can use both options at the same time.
+
+- Exclude category of the context item - excludes items in the recommendation response from the same categories as the context item in the recommendation request. Select this checkbox if you want do not want to recommend a client items from the category of the currently rendered item. For example, 
+if customer browses items from category *TV*, you can set that other recommendation box or boxes will not display recommendations from this category. This is automatically defined from the context.
```
Comment (DominikaK) [inline docs/personalization/filters.md:34]:
```suggestion
if the customer is browsing items from the *TV* category, you can ensure that recommendation boxes do not display recommendations from this category. This is automatically defined from the context.
```

File: docs/personalization/filters.md (hunk 5 above)
Comment (DominikaK) [reply docs/personalization/filters.md:33]: or
```suggestion
- **Exclude category of the context item** - excludes items in the recommendation response from the same categories as the context item in the recommendation request.
Select this checkbox if you do not want to recommend items from the category of the currently rendered item. For example, 
```
if it is a label in the UI

File: docs/personalization/filters.md [hunk 7]
```diff
@@ -22,6 +22,20 @@ User profile-based filters are applicable in both publishing and eCommerce use c
...
+
+- Exclude category - use to define lists of categories which should not be recommended in a scenario. Excludes the category that is last in the path. For example, in the following category path: *Furniture/Living room/Sofas*, for this scenario all items from *Sofas* category are excluded. You can add many categories.
```
Comment (DominikaK) [inline docs/personalization/filters.md:36]:
```suggestion
- **Exclude category** - define lists of categories which should not be recommended in a scenario.
Excludes the category that is last in the path.
For example, in the following category path: *Furniture/Living room/Sofas*, for this scenario all items from the *Sofas* category are excluded. You can add many categories.
```

File: docs/personalization/scenarios.md [hunk 8]
```diff
@@ -51,15 +51,13 @@ The configuration performed in this step is applied only to the selected scenari
 
 ## Filter configuration
 
-For every recommendation scenario a set of general filters can be defined that 
-are applied to all recommendations from every model used in the scenario.
-
-![General filters](img/scenario_filters.png "General filters")
+You can define a set general filters for every recommendation scenario.
+Filters are applied to all recommendations come from models selected in the strategy.
```
Comment (DominikaK) [inline docs/personalization/scenarios.md:55]:
```suggestion
Filters are applied to all recommendations that come from models selected in the strategy.
```

File: docs/personalization/scenarios.md [hunk 9]
```diff
@@ -51,15 +51,13 @@ The configuration performed in this step is applied only to the selected scenari
...
 
 For more information, see [General filters](filters.md#general-filters).
 
-For each of the categories from the strategy configuration matrix, you can click 
-the **Configure** icon and configure category filters. 
+If you want to configure category filters for each of the categories from the strategy configuration matrix, click the **Configure** icon and make necessary changes.
```
Comment (DominikaK) [inline docs/personalization/scenarios.md:59]:
```suggestion
If you want to configure category filters for each of the categories from the strategy configuration matrix,
click the **Configure** icon and make necessary changes.
```

**Review by DominikaK — CHANGES_REQUESTED** [review]

**Pull request merged**